pallets/proofs-dealer/src/lib.rs

```diff
@@ -596,8 +597,18 @@ pub mod pallet {
 
             Self::do_submit_proof(&provider, &proof)?;
 
+            // Get last tick for which the submitter submitted a proof.
+            let last_tick_proven = match LastTickProviderSubmittedAProofFor::<T>::get(provider) {
+                Some(tick) => tick,
+                None => return Err(Error::<T>::NoRecordOfLastSubmittedProof.into()),
+            };
```

Comment:
I would prefer it if `do_submit_proof` returns the `last_tick_proven`. We avoid checking for an impossible error, and keep this logic out of `lib.rs`